runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -460,3 +461,158 @@ J9::ARM64::TreeEvaluator::ArrayStoreCHKEvaluator(TR::Node *node, TR::CodeGenerat
 
    return NULL;
    }
+
+static TR::Register *
+genCAS(TR::Node *node, TR::CodeGenerator *cg, TR::Register *objReg, TR::Register *offsetReg, TR::Register *oldVReg, TR::Register *newVReg, 
+      TR::LabelSymbol *doneLabel, TR::Node *objNode, int32_t oldValue, bool oldValueInReg, bool isLong, bool casWithoutSync = false)
+   {
```

Comment:
@0xdaryl Thank you for the review! Looking at P's again, I understand the purpose of the loop. Just to clarify, should the instructions I have here in `genCAS()` that I was trying to use to do the CAS part be removed/commented out, and replaced with an assert? I can also introduce the loop in my changes.

Reply:
Most (all?) of the Power implementation doesn't apply to AArch64 as the swap semantics are different.  If you want to implement the "shell" (including the loop) in the absence of an actual CAS instruction implementation, go ahead.  Otherwise, if you want to collapse this whole function into an assert that's fine too.